hail/python/test/hail/table/test_table.py

```diff
@@ -348,57 +348,42 @@ def test_semi_anti_join(self):
         assert ht.semi_join(ht2).count() == 3
         assert ht.anti_join(ht2).count() == 7
 
-    def test_joins(self):
-        kt = hl.utils.range_table(1).key_by().drop('idx')
+    def test_indirected_joins(self):
+        kt = hl.utils.range_table(1)
         kt = kt.annotate(a='foo')
 
-        kt1 = hl.utils.range_table(1).key_by().drop('idx')
+        kt1 = hl.utils.range_table(1)
         kt1 = kt1.annotate(a='foo', b='bar').key_by('a')
 
-        kt2 = hl.utils.range_table(1).key_by().drop('idx')
+        kt2 = hl.utils.range_table(1)
         kt2 = kt2.annotate(b='bar', c='baz').key_by('b')
 
-        kt3 = hl.utils.range_table(1).key_by().drop('idx')
+        kt3 = hl.utils.range_table(1)
         kt3 = kt3.annotate(c='baz', d='qux').key_by('c')
 
-        kt4 = hl.utils.range_table(1).key_by().drop('idx')
+        kt4 = hl.utils.range_table(1)
         kt4 = kt4.annotate(d='qux', e='quam').key_by('d')
 
-        ktr = kt.annotate(e=kt4[kt3[kt2[kt1[kt.a].b].c].d].e)
-        self.assertTrue(ktr.aggregate(agg.collect(ktr.e)) == ['quam'])
-
-        ktr = kt.select(e=kt4[kt3[kt2[kt1[kt.a].b].c].d].e)
-        self.assertTrue(ktr.aggregate(agg.collect(ktr.e)) == ['quam'])
-
-        self.assertEqual(kt.filter(kt4[kt3[kt2[kt1[kt.a].b].c].d].e == 'quam').count(), 1)
+        assert kt.aggregate(agg.collect(kt4[kt3[kt2[kt1[kt.a].b].c].d].e)) == ['quam']
 
+    def test_table_matrix_join_combinations(self):
         m = hl.import_vcf(resource('sample.vcf'))
         vkt = m.rows()
-        vkt = vkt.select(vkt.qual)
-        vkt = vkt.annotate(qual2=m.index_rows(vkt.key).qual)
-        self.assertTrue(vkt.filter(vkt.qual != vkt.qual2).count() == 0)
+        assert vkt.filter(vkt.qual != m.index_rows(vkt.key).qual).count() == 0
 
-        m2 = m.annotate_rows(qual2=vkt.index(m.row_key).qual)
-        self.assertTrue(m2.filter_rows(m2.qual != m2.qual2).count_rows() == 0)
+        assert m.filter_rows(m.qual != vkt.index(m.row_key).qual).count_rows() == 0
 
-        m3 = m.annotate_rows(qual2=m.index_rows(m.row_key).qual)
-        self.assertTrue(m3.filter_rows(m3.qual != m3.qual2).count_rows() == 0)
+        assert m.filter_rows(m.qual != m.index_rows(m.row_key).qual).count_rows() == 0
 
-        kt5 = hl.utils.range_table(1).annotate(key='C1589').key_by('key')
-        m4 = m.annotate_cols(foo=m.s[:5])
-        m4 = m4.annotate_cols(idx=kt5[m4.foo].idx)
+        kt5 = hl.utils.range_table(1).annotate(s='C1589').key_by('s')
+        m2 = m.annotate_cols(foo=m.s[:5])
         n_C1589 = m.filter_cols(m.s[:5] == 'C1589').count_cols()
```

Comment:
confusing to define `m2` but not use it for a couple lines

Reply:
addressed